iron-form.html

```diff
@@ -456,6 +457,20 @@
 
         this.fire('iron-form-reset');
       }, 1);
+    },
+
+    _isChildOfRegisteredParent: function(node) {
+      var parent = Polymer.dom(node).parentNode;
```

Comment:
I'm not sure that's better? Right now, for every element in `customElements` we call `_isChildOfRegisteredParent`, which looks at its parents, so we have `O(n*p)`, where `n` is the number of elements in `customElements`, and `p` is the number of parents (usually 1 or very small). With your change, `_isChildOfRegisteredParent` would also do a loop of all the elements, so it would be `O(n*n)` which is imo significantly larger than `O(n*p)`.